open-craft/ansible-playbooks
# [BB-3036] Added new PPA for MySQL 5.6 support

The old PPA we have added to our xenial instance was gone,
this PPA is coming from ubuntu's earlier releases where MySQL 5.6
was supported.

**JIRA tickets**: [BB-3036](https://tasks.opencraft.com/browse/BB-3036)

~~**Discussions**: Link to any public dicussions about this PR or the design/architecture. Otherwise omit this.~~

~~**Dependencies**: None~~

~~**Screenshots**: Always include screenshots if there is any change to the UI.~~

~~**Sandbox URL**: TBD - sandbox is being provisioned.~~

~~**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.~~

**Testing instructions**:

1. Make a directory for this vargrant box, "mkdir ansible_exp" and "cd ansible_exp"
2.  Do `vagrant init ubuntu/xenial64`
~3. `vagrant init`~
4.  Clone this repot and switch branch to `farhaan/bb-3036-fix-mysql-deployment-playbook` inside this make sure to run ``git submodule update --init --recursive`` to get the modules.
5.  We need to add sync_folder in vagrant file, do vim Vagrantfile and add the following line 
``
config.vm.synced_folder "/path-to/your/ansible-playbooks", "/ansible-playbooks"
`` 
Here add replace the first parameter with your cloned repo path
and edit the files as mentioned in the diff below

[Ansible Diff](https://gist.github.com/farhaanbukhsh/890a1cb8d74a3ac790efc8319d49f80a)

and create an inventory file you can do `vim inventory` and put 

```
[mysql]
127.0.0.1 ansible_connection=local

```
in the file

6. ``vagrant up``
7. ``vagrant ssh``
8. ``cd  /ansible-playbook``
9. If you are using a machine with low spec you can uncomment the ``mysql_innodb_buffer_pool_size`` variable else it should be fine
10. sudo apt update  &&  sudo apt upgrade
11. sudo apt install python3 python3-pip build-essential libssl-dev libffi-dev python-dev -y
12. export LC_ALL=C
13.  pip3 install -r requirements.txt
14. I am going to run two roles separately for easy debugging you can run in one shot too
15.`` cd  playbooks``
16. ``ANSIBLE_RETRY_FILES_ENABLED=0 ansible-playbook -vvv mysql.yml --tags 'mysql' -i ../inventory``
17. ``ANSIBLE_RETRY_FILES_ENABLED=0 ansible-playbook -vvv mysql.yml --tags 'geerlingguy.mysql' -i ../inventory``
18. Above both commands should go clean and if you do `mysql --version` command you should see `5.6.16`
19. Running ``systemctl status mysql.service`` should also show you that it's active. 


**Author notes and concerns**:

This enables the old trusty thar repo, I think this is fine since we are anyhow using outdated versions,
I feel this method should work on 20.04 as well if I am not wrong I haven't tested that yet though.

**Reviewers**
